packages/vizbuilder/src/helpers/chartconfig.js

```diff
@@ -0,0 +1,330 @@
+    const {drilldown, measure} = query;
+
+    const drilldownName = drilldown.name;
+    const measureName = measure.name;
+
+    const config = {
+      ...commonConfig,
+      xConfig: {title: null},
+      y: measureName,
+      yConfig: {title: measureName},
+      groupBy: drilldownName
+    };
+
+    return config;
+  },
+  geomap(commonConfig, query, flags) {
+    const drilldownName = query.drilldown.name;
+
+    const config = {
+      ...commonConfig,
+      ...flags.colorScale,
+      tiles: false,
+      id: `ID ${drilldownName}`,
+      groupBy: `ID ${drilldownName}`,
+      ocean: "transparent",
+      projection: "geoAlbersUsa",
+      topojson: "/topojson/states.json",
+      topojsonId: "id",
+      topojsonKey: "states",
+      zoom: true,
+      zoomFactor: 2,
+      zoomScroll: false
+    };
+
+    if (flags.activeType === "geomap") {
+      const size = getAreaChartDimensions();
+      config.colorScaleConfig = {
+        scale: "jenks",
+        height: size.height,
+        width: size.width
+      };
+    }
+
+    return config;
+  },
+  lineplot(commonConfig, query) {
+    const {timeDrilldown, drilldown, measure, moe} = query;
+
+    const drilldownName = timeDrilldown.name;
+    const measureName = measure.name;
+
+    const config = {
+      ...commonConfig,
+      discrete: "x",
+      groupBy: drilldown.hierarchy.levels[1].name,
+      x: drilldownName,
+      xConfig: {title: drilldownName},
+      y: measureName,
+      yConfig: {title: measureName}
+    };
+
+    if (moe) {
+      const moeName = moe.name;
+
+      config.confidence = [
+        d => d[measureName] - d[moeName],
+        d => d[measureName] + d[moeName]
+      ];
+      config.confidenceConfig = {
+        fillOpacity: 0.15
+      };
+    }
+
+    return config;
+  },
+  stacked(commonConfig, query) {
+    const config = this.lineplot(commonConfig, query);
+    return config;
+  },
+  treemap(commonConfig, query) {
+    const {drilldown, measure} = query;
+
+    const drilldownName = drilldown.name;
+    const measureName = measure.name;
+
+    const config = {
+      ...commonConfig,
+      groupBy: drilldownName,
+      y: measureName,
+      yConfig: {title: measureName}
+    };
+
+    return config;
+  }
+};
+
+export default function createChartConfig({
+  activeType,
+  query,
+  availableKeys,
+  year
+}) {
+  const availableCharts = new Set(
+    activeType ? [activeType] : Object.keys(charts)
+  );
+
+  const measureName = query.measure.name;
+  const getMeasureName = d => d[measureName];
+
+  const commonConfig = {
+    height: activeType ? 500 : 400,
+    legend: false,
+
+    tooltip: true,
+    tooltipConfig: tooltipConfig(query),
+
+    duration: 0,
+    loadingMessage: "Loading",
+    sum: getMeasureName,
+    value: getMeasureName,
+    totalConfig: {
+      fontSize: 14
+    }
+  };
+
+  const colorScale = {
+    colorScale: measureName,
+    colorScalePosition: "bottom",
+    colorScaleConfig: {width: 0, height: 0}
+  };
```

Comment:
I think we should remove this extra config object, don't pass it through `flags`, and just allow each visualization config to set the `colorScale: measureName`.

`colorScalePosition` should be set by the user, and `colorScaleConfig` should not have to be set anymore (I fixed `colorScalePosition: false`)

Reply:
removed in 096b8f5a0900b84d4341cca21e35db479343399a